src/main/java/study1/racing/moveStrategy/MoveStrategy.java

```diff
@@ -0,0 +1,5 @@
+package study1.racing.moveStrategy;
+
+public interface MoveStrategy {
+  public int move(int maxDistance);
+}
```

Comment:
좀 더 `캡슐화`하여 생각해서 int를 리턴하기 보다는
`전진 가능하지 여부`를 리턴하도록 하는게 좋겠네요.

```suggestion
@FunctionalInterface
public interface MoveRule {
    boolean isAbleToMove();
}
```